collect_app/src/androidTest/java/org/odk/collect/android/formentry/FormHierarchyTest.java

```diff
@@ -0,0 +1,143 @@
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(0, R.id.primary_text))
+                .check(matches(withText("How many guests are in your party?")));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(0, R.id.secondary_text))
+                .check(matches(withText("2")));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(1, R.id.primary_text))
+                .check(matches(withText("Please provide details for each guest.")));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(2, R.id.primary_text))
+                .check(matches(withText("Guest details")));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(2, R.id.secondary_text))
+                .check(matches(withText("Repeatable Group")));
+        FormEntry.clickOnText("Guest details");
+        onView(withId(R.id.list)).check(matches(RecyclerViewMatcher.withListSize(R.id.list, 2)));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(0, R.id.primary_text))
+                .check(matches(withText("Guest details > 1")));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(1, R.id.primary_text))
+                .check(matches(withText("Guest details > 2")));
+        FormEntry.clickJumpStartButton();
+        FormEntry.putText("1");
+        FormEntry.clickGoToIconInForm();
+        FormEntry.clickOnText("Guest details");
+        onView(withId(R.id.list)).check(matches(RecyclerViewMatcher.withListSize(R.id.list, 1)));
+        onView(withRecyclerView(R.id.list)
+                .atPositionOnView(0, R.id.primary_text))
+                .check(matches(withText("Guest details > 1")));
+    }
+
+    @Test
+    //https://github.com/opendatakit/collect/issues/2936
+    public void test3() {
+        MainMenu.startBlankForm("formHierarchy3");
+        FormEntry.swipeToNextQuestion();
+        FormEntry.swipeToNextQuestion();
+        FormEntry.swipeToNextQuestion();
+        FormEntry.swipeToNextQuestion();
+        FormEntry.swipeToNextQuestion();
+        FormEntry.swipeToNextQuestion();
+        FormEntry.clickOnString(R.string.add_another);
+        FormEntry.swipeToNextQuestion();
+        FormEntry.clickOnString(R.string.add_another);
+        FormEntry.swipeToNextQuestion();
+        FormEntry.clickOnString(R.string.add_repeat_no);
+        FormEntry.clickOnString(R.string.add_repeat_no);
+        FormEntry.clickGoToIconInForm();
+        onView(withId(R.id.list)).check(matches(RecyclerViewMatcher.withListSize(R.id.list, 3)));
+        FormEntry.clickOnText("Group 1");
+        onView(withId(R.id.list)).check(matches(RecyclerViewMatcher.withListSize(R.id.list, 3)));
+        FormEntry.checkIfTextDoesNotExist("Repeat Group 1");
+    }
+
+    @Test
+    //https://github.com/opendatakit/collect/issues/2942
+    public void test4() {
```

Comment:
I think it would be nice for these tests to be named around what feature they are driving out. For instance could this one be `canDeleteLastGroup_whenInNestedRepeat`? It's definitely nice to have the github links but just thinking through if one of these tests broke it would be nicer (for me personally) to be able to get a sense of what this test was doing quickly.

Reply:
Improved.